alicloud/data_source_alicloud_zones.go

```diff
@@ -78,15 +78,15 @@ func dataSourceAlicloudZonesRead(d *schema.ResourceData, meta interface{}) error
 
 	var zoneTypes []ecs.ZoneType
 	for _, types := range resp {
-		if insType != "" && !constraints(types.AvailableInstanceTypes.InstanceTypes, insType) {
+		if len(types.AvailableInstanceTypes.InstanceTypes) == 0 || insType != "" && !constraints(types.AvailableInstanceTypes.InstanceTypes, insType) {
```

Comment:
The logical of line 81 is confused, and is it should be "if len(types.AvailableInstanceTypes.InstanceTypes) == 0 || ( insType != "" && !constraints(types.AvailableInstanceTypes.InstanceTypes, insType) )" ? Right?

Same the next code.